app/routes.rb

```diff
@@ -35,32 +20,38 @@ def with_json_body
 end
 
 get '/guides/:id/raw' do
-  guides.find(id: params['id']).projection(_id: 0).to_a.first.to_json
+  GuideCollection.find(params['id']).raw.to_json
 end
 
 get '/guides/:id' do
-  guides.find(id: params['id']).projection(_id: 0).map {|it| GitIo::Guide.new(it) }.to_a.first.to_json
+  GuideCollection.find(params['id']).to_json
+end
+
+get '/guides/:organization/:repository/raw' do
+  GuideCollection.find_by_slug(params['organization'], params['repository']).raw.to_json
+end
+
+get '/guides/:organization/:repository' do
+  GuideCollection.find_by_slug(params['organization'], params['repository']).to_json
```

Comment:
But what could you extract from it?

You should have look at the original version :stuck_out_tongue: